File: src/gestorAplicacion/Salas/Sala2D.java
package gestorAplicacion.Salas;
import java.io.Serializable;
import java.util.*;
import gestorAplicacion.Autocine.*;

public class Sala2D extends Sala implements Serializable{
	// Serializacion.
		private static final long serialVersionUID = 1L;
		static List<Sala2D> salas2D;
		static {
			salas2D= new ArrayList<Sala2D>();
		}
	
	public Sala2D(int filas, int columnas, int filasPreferencial, Autocine autocine) {
		super(filas, columnas, filasPreferencial, 2000, autocine);
	}
	
	public Sala2D(int preferencial, Autocine autocine) {
		this(8, 12, preferencial, autocine);
	}
	
	public int cantidadPuestos() {
	/*No recibe nada y devuelve un entero el cual corresponde a la cantidad de puestos
	disponibles*/

		return super.puestos.size();
	}
	
	public void crearPuestos() {			
	/*No recibe ningun parametro y no retorna nada
	Crea cada puesto según la cantidad de filas prefencial, filas, y columnas*/
		
		int total = filas*columnas; 	//numero de puestos

		int totalpreferencial = filasPreferencial*columnas; //numero de puestos prefe y se le resta 1 cada que se compre uno
													
		String tipo = "PREFENCIAL";
		
		for(int i = 0;i<total;i++) {	//for que itera la cantidad de puestos 
			
		   if(totalpreferencial<=0) {	//si se acabaron los preferencial cambia tipo a GENERAL
			 tipo = "GENERAL";
			}
		   else {				//sino  se reduce totalpreferencial en uno
			 totalpreferencial--;
			}
			
		   Puesto puesto = new Puesto(tipo,i+1);	
											
		   puestos.add(puesto);					
		}
	}
	
	//get y set
	public static List<Sala2D> getSalas2D() {
		return salas2D;
	}
	public static void setSalas2D(List<Sala2D> salas2d) {
		salas2D = salas2d;
	}

}

File: Python/gestorAplicacion/Salas/Sala2D.py
from gestorAplicacion.Autocine import *

#*
# * @author Jimena Uribe Giraldo
# * @param Clase Sala2D.
# * @summary Clase que que hereda de la Clase Sala. En esta se crean los puestos para la Sala 2D.
# 

class Sala2D(Sala):
    # Serializacion.
    _SERIALVERSIONUID = 1
    salas2D = None
    @staticmethod
    def _static_initializer():
        gestorAplicacion.Salas.Sala2D.salas2D= []

    _static_initializer()

#JAVA TO PYTHON CONVERTER TODO TASK: There is no Python equivalent to multiple constructors:
#ORIGINAL LINE: public Sala2D(int filas, int columnas, int filasPreferencial, Autocine autocine)
    def __init__(self, filas, columnas, filasPreferencial, autocine):
        super().__init__(filas, columnas, filasPreferencial, 2000, autocine)

#JAVA TO PYTHON CONVERTER TODO TASK: There is no Python equivalent to multiple constructors:
#ORIGINAL LINE: public Sala2D(int preferencial, Autocine autocine)
    def __init__(self, preferencial, autocine):
        self(8, 12, preferencial, autocine)

    def cantidadPuestos(self):
        #	No recibe nada y devuelve un entero el cual corresponde a la cantidad de puestos
        #	disponibles

        return len(self.puestos)

    def crearPuestos(self):
        #	No recibe ningun parametro y no retorna nada
        #	Crea cada puesto según la cantidad de filas prefencial, filas, y columnas

        total = self.filas*self.columnas #numero de puestos

        totalpreferencial = self.filasPreferencial*self.columnas #numero de puestos prefe y se le resta 1 cada que se compre uno

        tipo = "PREFENCIAL"

        for i in range(0, total):

            if totalpreferencial<=0:
                tipo = "GENERAL"
            else:
                totalpreferencial -= 1

            puesto = Puesto(tipo,i+1)

            self.puestos.append(puesto)

    #get y set
    @staticmethod
    def getSalas2D():
        return gestorAplicacion.Salas.Sala2D.salas2D
    @staticmethod
    def setSalas2D(salas2d):
        gestorAplicacion.Salas.Sala2D.salas2D = salas2d


